### PR TITLE
Add geographical offset to calculated substations when same neighborhood

### PR DESCRIPTION
**Pull request by Meklo**

### CODE REVIEW

File: src/main/java/org/gridsuite/geodata/server/GeoDataService.java [hunk 1]
```diff
@@ -149,20 +151,31 @@ private void calculateMissingGeoData(Network network, List<Substation> substatio
...
     }
 
+    private double nextNeighbourhoodOffset(double neighborhoodOffset) {
+        return neighborhoodOffset > 0 ? (neighborhoodOffset * -1) : (neighborhoodOffset * -1) + CALCULATED_NEIGHBORHOOD_OFFSET;
+    }
+
     private void step(Step step, Network network, Map<String, Set<String>> sortedNeighbours, Map<String, SubstationGeoData> substationsGeoData,
                       Set<String> substationsToCalculate) {
+
+        Map<String, Double> calculatedSubstationsOffset = new HashMap<>();
```
Comment (geofjamg) [inline src/main/java/org/gridsuite/geodata/server/GeoDataService.java:161]: Why not using the Set<String> as a key, you would independent of order?

**Review by geofjamg — CHANGES_REQUESTED** [review]

File: src/main/java/org/gridsuite/geodata/server/GeoDataService.java (hunk 1 above)
Comment (Meklo) [reply src/main/java/org/gridsuite/geodata/server/GeoDataService.java:161]: That's what I did initially then switched it to a String for the following reasons : I reckon it takes less memory space, seemed a bit more optimized to check pre existing neighborhood configuration and the String is itself generated from a Set so the order would, if i'm not mistaken, always be the same.
Still I can roll it back if it's more robust with a Set as key

**Review by geofjamg — APPROVED** [review]

**Review by laurgarn — APPROVED** [review]
Test OK, with split of the two CUBNE7xDONZA lines at two different new substations

**Review by niconoir — APPROVED** [review]
Reviewed and tested

**Pull request merged**